Userland/which.cpp

```diff
@@ -0,0 +1,32 @@
+#include <AK/AKString.h>
+#include <AK/ScopedValueRollback.h>
+#include <AK/Vector.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+int main(int argc, char** argv)
+{
+    if (argc < 2) {
+        printf("usage: which [executable]\n");
+        return 0;
+    }
+
+    char* filename = argv[1];
+
+    String path = getenv("PATH");
+    if (path.is_empty())
+        path = "/bin:/usr/bin";
+
+    auto parts = path.split(':');
+    for (auto& part : parts) {
+        auto candidate = String::format("%s/%s", part.characters(), filename);
+        if(access(candidate.characters(), X_OK) == 0) {
+            printf("%s\n", candidate.characters());
+            break;
+        }
+    }
+
+    return 0;
```

Comment:
This should be `return 1;` to signal to the outside world that we didn't find a matching executable.